### PR TITLE
added fit by group table

### PR DESCRIPTION
**Pull request by LSLindeloo**
https://github.com/jasp-stats/jasp-issues/issues/711 (fixed for SEM)
fixes https://github.com/jasp-stats/jasp-issues/issues/2690
also adds chi^2 diff tests for multiple models and groups

### CODE REVIEW

File: R/sem.R [hunk 1]
```diff
@@ -462,33 +462,76 @@ checkLavaanModel <- function(model, availableVars) {
...
+    grouptab$addColumnInfo(name = "Chisq",    title = gettext("&#967;&sup2;"),       type = "number" ,
+                         overtitle = gettext("Baseline test"))
+    grouptab$addColumnInfo(name = "Df",       title = gettext("df"),                 type = "integer",
+                         overtitle = gettext("Baseline test"))
+    grouptab$addColumnInfo(name = "PrChisq",  title = gettext("p"),                  type = "pvalue",
+                         overtitle = gettext("Baseline test"))
+    if (length(options[["models"]]) > 1) {
+      grouptab$addColumnInfo(name = "dchisq",   title = gettext("\u0394\u03C7\u00B2"), type = "number" ,
+                           overtitle = gettext("Difference test"))
+      grouptab$addColumnInfo(name = "ddf",      title = gettext("\u0394df"),           type = "integer",
```
Comment (Kucharssim) [inline R/sem.R:484]: here we only want to translate the "df"
```suggestion
      grouptab$addColumnInfo(name = "ddf",      title = gettextf("%1$sdf", "\u0394"),           type = "integer",
```

**Review by Kucharssim — CHANGES_REQUESTED** [review]
looks great, just a couple of comments

**Comment by juliuspfadt** [comment]
Screenshot 2024-02-21 at 16 32 27

**Comment by juliuspfadt** [comment]
It also seems very annoying that we have to fit the model again for each group, if only lavaan would give us the AIC and BIC

File: R/sem.R [hunk 2]
```diff
@@ -540,13 +568,97 @@ checkLavaanModel <- function(model, availableVars) {
...
+            if(equalityConstraints[j, "lhs"] %in% syntax_group[, "plabel"]) {
+              syntax_group <- rbind(syntax_group, equalityConstraints[j,])
+            }
+          }
+        }
+        syntax_group[["group"]] <- syntax_group[["block"]] <- rep(1, nrow(syntax_group))
+        lav_args[["model"]] <- syntax_group
+        dataset_group <- dataset[dataset[[options[["group"]]]] == groupNames[group],]
+        lav_args[["data"]]  <- dataset_group
+        fit_group <- try(do.call(lavaan::lavaan, lav_args))
```
Comment (juliuspfadt) [inline R/sem.R:609]: @Kucharssim, this code does not work when there are equality constraints, which is not surprising. Lorenzo fits the model for the subgroup only, but with invariance testing equality constraints are between the two groups, e.g., the loadings of group1 are equal to the loadings of group2. It seems to me that the equality constraints do not matter for the fit of each group, so we should estimate the model with free parameters unless some the constraints are not between groups but within a group. I wonder how we would find distinguish here though....

**Comment by juliuspfadt** [comment]
Okay, this should be ready as well. this is what I eventually settled on: 
Screenshot 2024-02-23 at 14 37 53
I actually think it makes little sense to get the fit per group cause giving a grouping variable is not the same as fitting one model per group. The fit function is minimized for the complete model with a group variable. so the fit of the groups is the same as the fit of the complete model.

**Comment by juliuspfadt** [comment]
now like this: 
Screenshot 2024-04-30 at 19 36 39

**Comment by juliuspfadt** [comment]
I suppose I should've checked, weird that I didn't notice the test failing. It's caused by imposing meanstructure but then not fixing any intercepts to 0. I wonder if in qml there is an option for something like a triple of checkboxes, they can be all checked but not unchecked simultaneously. Suppose .quitAnalysis will have to do for now.

**Pull request merged**